tests/Unit/Http/Requests/Album/MoveAlbumsRequestTest.php

```diff
@@ -23,11 +23,22 @@ class MoveAlbumsRequestTest extends BaseRequestTest
 {
 	public function testAuthorization()
 	{
+		$album = $this->createUserAndAlbum();
 		Gate::shouldReceive('check')
-			->with(AlbumPolicy::CAN_EDIT, [AbstractAlbum::class, null])
+			->times(4) // TODO:// not only the times, but make sure the right arguments are passed, but this is hard to do find with mockery
 			->andReturn(true);
 
 		$request = new MoveAlbumsRequest();
+		$request->setContainer($this->app);
+		$request->setRedirector($this->app['redirect']);
+		$request->merge([
+			RequestAttribute::PARENT_ID_ATTRIBUTE => null,
+			RequestAttribute::ALBUM_IDS_ATTRIBUTE => [$album->id],
+			RequestAttribute::ALBUM_ID_ATTRIBUTE => $album->id,
+			RequestAttribute::ALBUM_IDS_ATTRIBUTE . '.*' => $album->id,
+		]);
+
+		$request->validateResolved(); // hydrate the request Class with the data before authorizing . Fighting the framework a bit
 
 		$this->assertTrue($request->authorize());
```

Comment:
No longer necessary, because this is called directly in `validateResolved()`

If `$request->authorize()` returns false, an exception is raised.

Reply:
weird class inheritance problem, 

if you don't call this `$request->validateResolved()` then the `processValidatedValues()` method will not get called. then 

`$this->album`
`$this->albums`

will not get initialized 😬 

then we get this error where it says `$this->albums` can;t be used before initializing it. 


but its a good thing, the logic inside `processValidatedValues()` will also gets covered with this setup, but all in all this setup looks a bit weird to me., (overriding the `processValidatedValues()` method) . even though right now the tests cover them logic

Reply:
>  (overriding the `processValidatedValues()` method) 

Understandable, but that is because Laravel does not resolve the values then Authorize, it first authorize then validates. Unfortunately, this also means that you would be double querying to validate whether a user can access an album: first by fetching the Id to validate ownership etc, then again to fetch the data...
Hence why we first fetch, then validate.